vm/src/org/mozartoz/truffle/nodes/builtins/BuiltinsManager.java

```diff
@@ -110,6 +110,20 @@ public static void defineBuiltins() {
 		BOOT_MODULES.put("Boot_WeakRef", BOOT_MODULES.get("Boot_WeakReference")); // TODO: hack
 	}
 
+	public static OzNode createNodeFromFactory(NodeFactory<? extends OzNode> factory, OzNode... args) {
+		Builtin builtin = factory.getNodeClass().getAnnotation(Builtin.class);
+		if (builtin == null) {
+			builtin = Builtin.DEFAULT;
+		}
+
+		Object[] arguments = new OzNode[args.length];
+		for (int i = 0; i < args.length; i++) {
+			arguments[i] = argumentNode(builtin, i, args[i]);
+		}
+
+		return factory.createNode(arguments);
+	}
```

Comment:
Actually, I modified readArgumentNode in order to use it. But I'm going to inline it directly in installBuiltins.